### ISSUE TITLE
[ExportVerilog] Consider to cache the result of `isExpressionUnableToInline`

### ISSUE THREAD

**Issue by uenoku** [llvm/circt#2489]
`isExpressionUnableToInline(op)` checks all users to determine the given operation is allowed to emit inline. 
I haven't done proper profiling but it might be possible to speed up ExportVerilog by cacheing the result of `isExpressionUnableToInline`.

**Comment by lattner** [comment]
Yeah this is a significant amount of compile time.  Instead of caching, is there a better algorithm to use here that isnt' quadratic?
